### PR TITLE
Use connection_pool as a dumb thread pool instead of Celluloid

### PR DESCRIPTION
**Pull request by foca**
Instead of depending on Celluloid, which is complex and way too much for what we actually nead, this sets up a ConnectionPool that checks out nothing (i.e. the connections are just `true`), so that we can abuse it as a thread pool.

This would _always_ make the workers threaded (set `DISC_CONCURRENCY=1` if you want a single thread...).

Having a single code path is good, though. It means it's simpler to test features / bugfixes as we don't need to run every test by duplicate on threaded/nonthreaded mode.

Oh, and speaking about tests, did I mention this is just a WIP? :P

cc @pote

### CODE REVIEW

File: lib/disc.rb [hunk 1]
```diff
@@ -21,7 +27,7 @@ def self.disque=(disque)
   end
 
   def self.disque_timeout
-    @disque_timeout ||= 100
+    @disque_timeout ||= Integer(ENV.fetch('DISQUE_TIMEOUT', 2000))
```
Comment (foca) [inline lib/disc.rb]: Not entirely sure why `Disc.disque_timeout` had a default different from `DISQUE_TIMEOUT`. Was this on purpose?

File: lib/disc.rb [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 
 require_relative 'disc/version'
 
 class Disc
-  attr_reader :disque,
-    :disque_timeout
+  def self.pool
+    @pool ||= begin
+      concurrency = Integer(ENV.fetch('DISC_CONCURRENCY', 25))
+      tolerance = Integer(ENV.fetch('DISC_TOLERANCE', disque_timeout / 1000.0))
```
Comment (foca) [inline lib/disc.rb]: The default timeout for each thread would be the same as for disque to consider a job was not finished. Maybe this should be lower? Dunno, seemed like a decent default.

Comment (pote) [inline lib/disc.rb]: Decent defaults are all we can hope for, if a reason to change it presents itself then we'll reevaluate, for now that's good enough :)

File: lib/disc.rb (hunk 1 above)
Comment (pote) [inline lib/disc.rb]: Probably not.

**Comment by pote** [comment]
I think dropping Celluloid and having a single code path are good things, however I'd like to move to something that won't break the non-threaded mode at least, what are your thoughts on the stability and safety of using `ConnectionPool`? Should we keep this as a WIP for the time being until we cook up a more definitive solution?
